### PR TITLE
UCP/DOC: Minor documentation fixes

### PR DESCRIPTION
**Pull request by brminich**
Would help to avoid confusion with rkeys
(someone forgot to release rkeys in his app)

### CODE REVIEW

**Comment by shamisp** [comment]
Useful clarification. @bbenton @tonycurtis please take a look.

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -1994,6 +1994,10 @@ void ucp_rkey_buffer_release(void *rkey_buffer);
  * Application code should not make any changes to the content of the RKEY
  * buffer.
  *
+ * @note The application is responsible to release RKEY when it is no longer
+ *       needed by calling the @ref ucp_rkey_destroy "ucp_rkey_destroy()"
```
Comment (bbenton) [inline src/ucp/api/ucp.h]: suggested re-wording:
responsible _**for releasing the RKEY object**_ when it ...

**Comment by bbenton** [comment]
👍

**Comment by shamisp** [comment]
👍 @tonycurtis ?

**Comment by tonycurtis** [comment]
Left a comment about an hour ago

**Comment by shamisp** [comment]
@tonycurtis - it was addressed. Let's merge.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9145/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9148/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9155/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9158/ for details (Mellanox internal link).

**Pull request merged**